JoinColony/colonyCDapp
# Feat: "Validate" tokens on token select and in manage tokens sagas

## Description

Currently token addresses are not validated when added them to the manage tokens form, meaning any properly formatted address (including colony addresses and user addresses) are accepted.

This PR adds extra validation to the form to prevent these values being added using recent changes to `fetchTokenFromChain`. It also adds the same checks to the manage tokens sagas as an extra precaution.

## Testing

* Enable the reputation voting extension
* Create a 'Manage tokens' action.
* Add a new token to the table. Try using an existing colony address, a user address, a properly formatted address (`0x0000000000000000000000000000000000000001`) these should all be rejected. You can also try a valid token from another colony - if doing this in Planex you will either have to switch to Wayne and add CREDS there, or first remove GGG and re-add it to test this.
* Create the motion and complete the flow to check the expected changes are applied.

## Diffs

**New stuff** ✨

* Token selector component checks if token is "valid"
* Manage tokens sagas "validate" tokens


Resolves #2687
